### DISCUSSION TITLE
hrldas.exe error: unrealistic LWDN and TV, TG at first timestep

### DISCUSSION THREAD

**Discussion by neha12-3** [NCAR/hrldas#174] [Q&A] [answered]
![1712904352448](https://github.com/NCAR/hrldas/assets/167417195/748842ed-19d1-4da7-baa6-524124934fe8)

**Comment by RAbolafiaRosenzweig** [comment]
I suspect the prescribed forcing for downward longwave radiation is not physically realistic. Please check the meteorological forcing inputs to confirm each met. variable is within physically realistic bounds

**Reply by neha12-3** [reply]
i used era 5 land surface hourly dat on single level from 1940 to present (longwave) i again download and done all the steps again but the issue cant get solved

**Reply by cenlinhe** [reply]
Did the model still crash at the first time step? If so, could you please check your generated forcing file at the first 2 time step for all 7 forcing variables to see if there are any unreasonable values? If your forcing file is not large, you could also attach it here so that we could also check it. Also, please provide your complete namelist.hrldas and the namelist used for create_forcing (i.e., namelist.input.ERA5).

**Reply by neha12-3** [reply]
i am not able t o upload my forcing field data here, as this is not taking .nc or grib files, can i forward u on your email
/media/neha/3C9AE25D9AE21366/remp/INIT
/media/neha/3C9AE25D9AE21366/remp/lw
/media/neha/3C9AE25D9AE21366/remp/lwclearsky
/media/neha/3C9AE25D9AE21366/remp/LWDOWNnew
/media/neha/3C9AE25D9AE21366/remp/Psurf
/media/neha/3C9AE25D9AE21366/remp/RAINRATE
/media/neha/3C9AE25D9AE21366/remp/sphum2m
/media/neha/3C9AE25D9AE21366/remp/swdownnew
/media/neha/3C9AE25D9AE21366/remp/Tair
/media/neha/3C9AE25D9AE21366/remp/U
/media/neha/3C9AE25D9AE21366/remp/V
/media/neha/3C9AE25D9AE21366/remp/zdata

**Reply by neha12-3** [reply]
/media/neha/3C9AE25D9AE21366/namelist.hrldas
/media/neha/3C9AE25D9AE21366/namelist.input

**Reply by cenlinhe** [reply]
you can email me your netcdf forcing file (only need first two time steps) and your netcdf hrldas_setup file (or wrfinput file) as well as your namelist.hrldas and namelist.input files. my email: cenlinhe@ucar.edu

**Comment by neha12-3** [comment]
Thnku so much ..will work on this

**Comment by neha12-3** [comment]
Thank you, I will do that.

**Comment by cenlinhe** [comment, marked as answer]
After checking the user's forcing file, the entire LWDOWN field in the forcing file is wrong with extremely large values (>10^5). Something is wrong with the HRLDAS_forcing step when generating the forcing files. Please double check if there are any error messages during that step.

**Reply by neha12-3** [reply]
there was a message while running create_forcing.exe for soil temp and soil mois-source model land/sea data not found, assuming a bitmap exists.(is this effecting my run). and i have downloaded long wave data from era 5 land hourly data on single level from 1940 to present. there is issue in data or in data preparation step?

**Reply by neha12-3** [reply]
LWDOWNnew-i have this data from era 5 hourly data on single level from 1940 to present -surface thermal radiation downward clear sky

**Reply by cenlinhe** [reply]
Did you follow our tutorial on these steps of generating ERA-5 forcing: https://github.com/NCAR/hrldas/blob/master/tutorial/Note5_Regional_modeling_ERA5-Land_forcing.ipynb

**Reply by neha12-3** [reply]
thnku sir for this but i have a query that in this tutorial era land hourl data from 1950 to present is mentioned but here in this we cant select more than 1 month at a time so if we want 5 year data it will be difficult to download 1-1 month data for 5 years for 8 parameters thats why i choose another one i.e era 5 hourly on single level from 1940 tp present bcz in that we can select all the months at a time. is that the raeson for error in my run as i have taken other one

**Reply by neha12-3** [reply]
thnku sir i have a query that in this tutorial era land hourl data from 1950 to present is mentioned but here in this we cant select more than 1 month at a time so if we want 5 year data it will be difficult to dwnlod for 8 parameters thats why i choose another one i.e era 5 hourly on single level from 1940 tp present bcz in that we can select all the months at a time. is that the raeson for error in my run.

**Reply by tslin2** [reply]
yes, you can use python API to download the whole ERA5-Land data.

here is an another example,

https://github.com/joaohenry23/Download_ERA5_with_python

and information from CDS website.

https://cds.climate.copernicus.eu/api-how-to
https://confluence.ecmwf.int/display/CKB/How+to+download+ERA5

**Reply by tslin2** [reply]
if you use single-level one with 0.25 degree (https://cds.climate.copernicus.eu/cdsapp#!/dataset/reanalysis-era5-single-levels?tab=overview), that should also work, but you need to revise the postprocessing script also use the corresponding Zfile and LANDSfile. As I recall, the downward long-wave and shortwave radiation, precipitation for the single-level ERA5 data is not accumulated, but please double check!

**Reply by neha12-3** [reply]
thnku sir i have resolved the issue, your help plays a great role .Thnks to github.

**Comment by neha12-3** [comment]
hii sir/madam can anyone please provide me the script or code by which i can split era5 data into yearly,monthly,daily and hourly in single step in cdo

**Reply by tslin2** [reply]
Hi,
please see 
https://code.mpimet.mpg.de/projects/cdo/embedded/index.html#x1-1500002.2.12
those commands are likely what you looked for in cdo

**Comment by neha12-3** [comment]
can you please provide me the single command with which i can split data file into year, month day and hour in single step

**Reply by cenlinhe** [reply]
I am not aware of such command. You will need a loop (e.g., in bash or perl script) to do it using nco command.

**Reply by neha12-3** [reply]
can u suggest bash script that help in splitting data from year to month days and hours in a single step

**Reply by tslin2** [reply]
try 
cdo splitmon[,format]  infile obase

cdo splithour[,format]  infile obase

https://code.mpimet.mpg.de/projects/cdo/embedded/index.html#x1-1500002.2.12

**Reply by tslin2** [reply]
or try
codes_split_file
https://confluence.ecmwf.int/display/ECC/codes_split_file

An example can be found here
https://github.com/NCAR/hrldas/blob/master/hrldas/HRLDAS_forcing/run/examples/ERA5/Prepare_ERA5-Land_Forcing.sh

**Comment by neha12-3** [comment]
Thanks, I'll check it out.

**Comment by neha12-3** [comment]
i have got this error while following steps for cmake installation (eecodes) after this command -cmake -DCMAKE_INSTALL_PREFIX=/opt/eccodes ../eccodes-2.35.0-Source
Feature JPG_LIBJASPER enabled
-- Feature JPG_LIBOPENJPEG enabled
-- Could NOT find NetCDF (missing: NetCDF_C_INCLUDE_DIR NetCDF_C_LIBRARY) 
-- eccodes FAILED to find OPTIONAL package NetCDF
-- Could NOT find package NetCDF required for feature NETCDF -- 
-- Feature NETCDF was not enabled (also not requested) -- following required packages weren't found: NetCDF
-- Could NOT find AEC (missing: AEC_LIBRARY AEC_INCLUDE_DIR) 
CMake Error at cmake/ecbuild_log.cmake:190 (message):
  CRITICAL - AEC library was not found.

  AEC (Adaptive Entropy Coding) provides the WMO GRIB CCSDS compression and
  decompression of data.  This is highly recommended from ecCodes >= 2.25.0

  To force the build without it, use -DENABLE_AEC=OFF
Call Stack (most recent call first):
  CMakeLists.txt:135 (ecbuild_critical)


-- Configuring incomplete, errors occurred!
See also "/home/neha/Downloads/build/CMakeFiles/CMakeOutput.log".
See also "/home/neha/Downloads/build/CMakeFiles/CMakeError.log".

**Comment by neha12-3** [comment]
hi sir/mam i got this error while running ./hrldas.exe this error seens at 20180125_18 timestep. can you plz help
![IMG_20241025_130708](https://github.com/user-attachments/assets/d5e499bb-c23e-4ef6-b50c-6aa75cf3445a)
![IMG_20241025_130708](https://github.com/user-attachments/assets/317ae3b0-8a35-4939-aa58-a912358cfa9c)

**Reply by cenlinhe** [reply]
This problem usually happens if you have unrealistic values in your forcing file over some grids/points or unrealistic values in your initial setup file. Is this error occurring after the model runs for a while or just the first timestep?

**Comment by neha12-3** [comment]
Occuring after a half month run

**Reply by cenlinhe** [reply]
OK, then please check your forcing variables to see if there are any unrealistic values. and then what model timestep (NOAH_TIMESTEP) are you using? please use a smaller model timestep, e.g., 900 (i.e., 15-min) or 1800 (30-min).

**Comment by neha12-3** [comment]
for hrldas run i.e ldasout, can i use gdas fnl 0.25*0.25 data (that we use for wrfinput file) nd  how many days this data is needed for hrldas

**Reply by cenlinhe** [reply]
yes you can use gdas. the number of days needed depend on your simulation period.

**Comment by neha12-3** [comment]
can you provide me single coomand of grads from which i can extract data at single timestep at different levels over average area

**Reply by tslin2** [reply]
do you mean grads?https://en.wikipedia.org/wiki/GrADS

**Comment by neha12-3** [comment]
![1752123728225](https://github.com/user-attachments/assets/cc3b579e-d033-464e-9bc1-f3c942eea585)
having this error while running wrf.exe with assimilated wrfinput
![1752123728246](https://github.com/user-attachments/assets/f9f78fe9-fcae-4c5e-9ef0-b70c52ec4444)

**Comment by neha12-3** [comment]
i was running hrldas for 5 years but now i found that i am having exact same ldasout file for 5 year. what changes should i make in namelist  so that i can get hourly ldasout exact
it seems like mt hrldas was running at the 1 same single step for whole 5 year

**Reply by tslin2** [reply]
try changing SPLIT_OUTPUT_COUNT = 1                                      !    Number of times per output file         [default = 1   ] 
in the namelist.hrldas

**Reply by neha12-3** [reply]
already using split output count 1

**Reply by neha12-3** [reply]
what should i use

**Reply by tslin2** [reply]
do you set KDAY to 1825 (365 days for 5 years) 
KDAY  !    Number of days to run                   [default = none]

**Comment by neha12-3** [comment]
what split output count should i use if i want to run hrldas for 5 years
and want hourly or daily ldasout

**Reply by tslin2** [reply]
if you want hourly file, the SPLIT_OUTPUT_COUNT should be 1
if you set to 24, each daily file should have 24 hour output

**Comment by neha12-3** [comment]
![Uploading 1753434312783.jpg…]()
![1753348774962](https://github.com/user-attachments/assets/7fec43df-95a8-4106-946e-b4599b255717)
getting this error at initial time while running hrldas.exe

**Reply by tslin2** [reply]
You may check your input forcing especially precipitation is missing data or not, looking like the prec is inf

**Reply by cenlinhe** [reply]
also check the landmask consistency with your forcing map. e.g., you need to at least have valid values for forcing over all land points (particularly for coastal points and islands).

**Reply by neha12-3** [reply]
how to check landmask consistency

**Reply by cenlinhe** [reply]
you can look at the "LANDMASK" variable in your HRLDAS_setup file and compare it with your forcing variables to see if grids with LANDMASK value of 1 all have valid forcing values (i.e., the forcing variable landmask is consistent with that in HRLDAS_setup file).

**Comment by neha12-3** [comment]
<img width="1366" height="768" alt="Screenshot from 2025-07-25 14-56-14" src="https://github.com/user-attachments/assets/c2d0c3ba-3fd6-4101-84a7-fa351524df2d" />
<img width="1366" height="768" alt="Screenshot from 2025-07-25 14-56-25" src="https://github.com/user-attachments/assets/5362153e-10e9-4ba2-8bd4-f12d33a946f4" />
these are the namelist i am using in hrldas

**Comment by neha12-3** [comment]
can i use gldas data as initial field data and era 5 hourly as forcing data fo hrldas run then which namelist.input can i use gldas or era because both have different code for variables

**Comment by neha12-3** [comment]
can i use gldas data as initial field data and era 5 hourly as forcing data fo hrldas run then which vtable can i use gldas or era because both have different code for variables and also i have remapped my era 5 data according to gldas but stil facing error

**Comment by neha12-3** [comment]
can u plz mention pbl scheme and radiation physics option that work best with hrldas(high resolution data assimilation system)in north western indian region

**Reply by cenlinhe** [reply]
this seems a question for WRF forum not HRLDAS. Also, different combinations of pbl and radiation options work differently according to specific regions and time periods. Oftentimes, you will have to test it yourself. You can also ask this question on the WRF forum.

**Comment by neha12-3** [comment]
i am getting same soil moisture output in all ldasout files except starting 3-4months of ldasout out. i have started run from from 2018010100 and for upto april i get varying soil moisture value but after that my soil moisture value become constant in all ldaout file for rest months and years

**Reply by cenlinhe** [reply]
1. you can check other output variables like EDIR and ETRAN to see if these values vary. 
2. What study domain do you have? 
3. Are you looking at SOIL_M or SOIL_W? soil moisture should be SOIL_M. 
4. Can you also try to start the simulation in a warm season not winter month to see if it makes any difference? 
5. Can you check if your forcing variables vary during your constant soil moisture periods?

**Reply by neha12-3** [reply]
other variables like soil temperature longwave thy are varying but soil_m vary only like this 0.1598, 0.1597, 0.1596 but obs vary like 0.15, 0.16, 0.17. i am using north-west indian domain and soil_m. simulation during pre_monsoon (april)
